File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum returns the square of the sum of every number 1..n
func SquareOfSum(n int) int {
	sum := (n*n + n) / 2
	return sum * sum
}

// SumOfSquares returns the sum of every number 1..n, squared
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference of the sum of squares and the square of sums
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: ocaml/difference-of-squares/difference_of_squares.ml
(* This is the triangle number formula, squared *)
let square_of_sum n =
  let sum = (n*n+n)/2 in
    sum*sum

(* square pyramidal number formula *)
let sum_of_squares n =
  (n*(n+1)*(2*n+1))/6

let difference_of_squares n =
    square_of_sum n - sum_of_squares n
